include/eve/detail/function/simd/riscv/combine.hpp

```diff
@@ -0,0 +1,65 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch.hpp>
+#include <eve/arch/riscv/rvv_common_masks.hpp>
+#include <eve/detail/abi.hpp>
+#include <eve/module/core/regular/if_else.hpp>
+
+namespace eve::detail
+{
+template<typename T, typename N>
+EVE_FORCEINLINE auto
+combine(rvv_ const&, wide<T, N> const& l, wide<T, N> const& h) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  constexpr size_t combined_vl = N::combined_type::value;
+  using that_t                 = wide<T, typename N::combined_type>;
+
+  if constexpr( combined_vl > expected_cardinal_v<T> )
+  {
+    that_t that;
+
+    that.storage().segments[0] = l;
+    that.storage().segments[1] = h;
+
+    return that;
+  }
+  else
+  {
+    auto           wider_l        = bit_cast(l, as<that_t> {});
+    auto           wider_h        = bit_cast(h, as<that_t> {});
+    constexpr auto shift_size     = N::value;
+    that_t         wider_h_placed = __riscv_vslideup(wider_h, wider_h, shift_size, combined_vl);
+    auto           mask_all_ones  = rvv_true<T, N>();
+    auto           wider_mask     = bit_cast(mask_all_ones, as<logical<that_t>> {});
+    return if_else(wider_mask, wider_l, wider_h_placed);
```

Comment:
Unless `slide` does something silly, you should be able to just or them instead of if/else, no?

Reply:
+1

Reply:
Missed this. What do you mean by slide? In EVE code I found only slide_pattern, but it seems it can not be used here.